src/data/mystery-encounters/encounters/fiery-fallout-encounter.ts

```diff
@@ -0,0 +1,250 @@
+      });
+      scene.time.delayedCall(1200, () => {
+        animation.playWithoutTargets(scene, 140, 150, 2);
+      });
+
+      return true;
+    })
+    .withTitle(`${namespace}:title`)
+    .withDescription(`${namespace}:description`)
+    .withQuery(`${namespace}:query`)
+    .withSimpleOption(
+      {
+        buttonLabel: `${namespace}:option:1:label`,
+        buttonTooltip: `${namespace}:option:1:tooltip`,
+        selected: [
+          {
+            text: `${namespace}:option:1:selected`,
+          },
+        ],
+      },
+      async (scene: BattleScene) => {
+        // Pick battle
+        const encounter = scene.currentBattle.mysteryEncounter;
+        setEncounterRewards(scene,
+          { fillRemaining: true },
+          null,
+          () => {
+            giveLeadPokemonCharcoal(scene);
+          });
+
+        encounter.startOfBattleEffects.push(
+          {
+            sourceBattlerIndex: BattlerIndex.ENEMY,
+            targets: [BattlerIndex.PLAYER],
+            move: new PokemonMove(Moves.FIRE_SPIN),
+            ignorePp: true
+          },
+          {
+            sourceBattlerIndex: BattlerIndex.ENEMY_2,
+            targets: [BattlerIndex.PLAYER_2],
+            move: new PokemonMove(Moves.FIRE_SPIN),
+            ignorePp: true
+          },
+          {
+            sourceBattlerIndex: BattlerIndex.ENEMY,
+            targets: [BattlerIndex.ENEMY],
+            move: new PokemonMove(Moves.QUIVER_DANCE),
+            ignorePp: true
+          },
+          {
+            sourceBattlerIndex: BattlerIndex.ENEMY_2,
+            targets: [BattlerIndex.ENEMY_2],
+            move: new PokemonMove(Moves.QUIVER_DANCE),
+            ignorePp: true
+          });
+        await initBattleWithEnemyConfig(scene, scene.currentBattle.mysteryEncounter.enemyPartyConfigs[0]);
+      }
+    )
+    .withSimpleOption(
+      {
+        buttonLabel: `${namespace}:option:2:label`,
+        buttonTooltip: `${namespace}:option:2:tooltip`,
+        selected: [
+          {
+            text: `${namespace}:option:2:selected`,
+          },
+        ],
+      },
+      async (scene: BattleScene) => {
+        // Damage non-fire types and burn 1 random non-fire type member
+        const encounter = scene.currentBattle.mysteryEncounter;
+        const nonFireTypes = scene.getParty().filter((p) => p.isAllowedInBattle() && !p.getTypes().includes(Type.FIRE));
+
+        for (const pkm of nonFireTypes) {
+          const percentage = DAMAGE_PERCENTAGE / 100;
+          const damage = Math.floor(pkm.getMaxHp() * percentage);
+          applyDamageToPokemon(scene, pkm, damage);
+        }
+
+        // Burn random member
+        const burnable = nonFireTypes.filter(p => isNullOrUndefined(p.status) || isNullOrUndefined(p.status.effect) || p.status?.effect === StatusEffect.BURN);
+        if (burnable?.length > 0) {
+          const chosenPokemon = burnable[randSeedInt(burnable.length - 1)];
```

Comment:
```suggestion
          const roll = randSeedInt(burnable.length);
          const chosenPokemon = burnable[roll];
```
Please separate those.
I'm also not sure if you should do `-1` since `randSeedInt(1)` rolls `0 - 0` and `randSeedInt(2)` rolls `0 - 1` (if I'm not mistaken

Reply:
you're right, the length - 1 was a brainfart.